src/rocksdb.rs

```diff
@@ -299,19 +299,28 @@ impl<D> DBIterator<D> {
         }
     }
 
-    pub fn kv(&self) -> Option<(Vec<u8>, Vec<u8>)> {
-        if self.valid() {
-            Some((self.key().to_vec(), self.value().to_vec()))
-        } else {
-            None
+    pub fn kv(&self) -> Result<Option<Kv>, String> {
```

Comment:
seems we can remove `kv()`, and call `key()` and `value()` separatley without `to_vec()`?

Reply:
I prefer to do it in an another PR.